src/charm.py

```diff
@@ -142,6 +169,53 @@ def _run_sql_migration(self, set_timeout: bool) -> None:
         stdout, _ = process.wait_output()
         logger.info(f"Executing automigration: {stdout}")
 
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
```

Comment:
This should probably set just before the `container.push` call, because in lots of cases we'll end up in waiting or blocked.

Reply:
With the latest updates hydra adds a layer and replans the container while in this status, so I think it's fine to keep it that way?